author_generator.rb

```diff
@@ -71,9 +71,15 @@ def write_author_index(author_dir, author)
     def write_author_indexes
       author_dirs_written = []
 
+      if Jekyll::VERSION < '3.0.0'
+        $post_collection = self.posts
```

Comment:
Can you replace the global variable with local variable and also use the naming appropriately? 

The name for the variable also can be something like docs, or post_docs to make it consistent.